apps/pipelines/views.py

```diff
@@ -226,6 +238,24 @@ def _get_assistant_url(assistant_id: int):
         OptionsSource.built_in_tools_config: BuiltInTools.get_tool_configs_by_provider(),
         OptionsSource.text_editor_autocomplete_vars_llm_node: PromptVars.get_all_prompt_vars(),
         OptionsSource.text_editor_autocomplete_vars_router_node: PromptVars.get_router_prompt_vars(),
+        OptionsSource.voice_provider_id: (
+            [_option(selected_voice_provider.id, selected_voice_provider.type)] if selected_voice_provider else []
+        ),
+        OptionsSource.synthetic_voice_id: sorted(
+            [
+                _option(
+                    voice.id,
+                    str(voice),
+                    voice.service.lower(),
+                    None,
+                    None,
+                )
+                | {"provider_id": voice.voice_provider_id}
+                for voice in synthetic_voices
+                if selected_voice_provider and voice.service.lower() == selected_voice_provider.type
```

Comment:
Why not do this filtering when we do the query? You could avoid this if statement entirely.